src/java.base/windows/classes/sun/nio/fs/WindowsFileSystemProvider.java

```diff
@@ -334,6 +334,20 @@ private void checkReadAccess(WindowsPath file) throws IOException {
                                 0L);
             fc.close();
         } catch (WindowsException exc) {
+            try {
+                if (exc.lastError() == ERROR_CANT_ACCESS_FILE &&
+                        isUnixDomainSocket(file))
+                {
+                    FileChannel fc = WindowsChannelFactory
+                        .newFileChannel(file.getPathForWin32Calls(),
+                                file.getPathForPermissionCheck(),
+                                Set.of(WindowsChannelFactory.OPEN_REPARSE_POINT),
+                                0L);
+                    fc.close();
```

Comment:
The new version looks a bit strange. As I read it, the attempt to open the file fails with ERROR_CANT_ACCESS_FILE so you then check test if the file is a socket file. That succeeds so we should be done. What is the reason for opening the file again?

Reply:
I added the test of the socket reparse tag to be 100% sure that we weren't returning false positives for other cases of ERROR_CANT_ACCESS_FILE. But, otherwise, I thought I had to open the file the same way as before to be sure that it is readable. If checking the reparse tag is sufficient then great, that simplifies the change.